### PR TITLE
Upstream up-to 2020-09-22

### PR DESCRIPTION
**Pull request by amilcarlucas**
These are the ones that do need careful one-by-one discussion and testing

### CODE REVIEW

**Comment by amilcarlucas** [comment]
Done the first rebase today already :)

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -2661,11 +2185,14 @@
       <entry value="40" name="MAV_SENSOR_ROTATION_ROLL_90_PITCH_315">
         <description>Roll: 90, Pitch: 315</description>
       </entry>
+      <entry value="41" name="MAV_SENSOR_ROTATION_ROLL_270_YAW_180">
```
Comment (tridge) [inline message_definitions/v1.0/common.xml:2188]: this conflicts with ardupilot use of 41. I'm pretty sure this is an alias of an existing one
aliasing two to one confuses the auto-orientation code

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by amilcarlucas** [comment]
@tridge thanks for the review, I removed the offending commit.

**Comment by peterbarker** [comment]
On Tue, 29 Sep 2020, Amilcar Lucas wrote:

> @tridge thanks for the review, I removed the offending commit.

@amilcarlucas could I suggest splitting these commits into two PRs - those 
that will be *completely* unobjectionable (like making the XML confirm to 
the schema, removing slugs.xml), and those that may be objected to 
(changing the shape of messages etc etc).

There are also patches in here which make a change and then revert it - 
why did you choose to do that?

**Comment by amilcarlucas** [comment]
@peterbarker can you point to the objectionable commits in your opinion?

I tried to squash as much as possible. Can you point to where a change gets reverted later?

**Comment by peterbarker** [comment]
On Mon, 5 Oct 2020, Amilcar Lucas wrote:

> I tried to squash as much as possible. Can you point to where a change gets reverted later?

Add MAV_MISSION_TOO_MANY_ITEMS
Remove MAV_MISSION_TOO_MANY_ITEMS

**Comment by amilcarlucas** [comment]
@peterbarker Good catch. Fixed.

File: message_definitions/v1.0/uAvionix.xml [hunk 2]
```diff
@@ -4,6 +4,7 @@
   <!-- company URL: http://www.uAvionix.com                       -->
   <!-- email contact: matt@uAvionix.com or jeff@uAvionix.com      -->
   <!-- mavlink ID range: 10000 - 10099                            -->
+  <include>common.xml</include>
```
Comment (auturgy) [inline message_definitions/v1.0/uAvionix.xml:7]: I don't think this is right.  Best check with uavionix, but this change didn't come from them

File: doc/mavlink_gitbook.py [hunk 3]
```diff
@@ -44,7 +44,7 @@
 
 Dialects may *include* other MAVLink XML files.
 A typical pattern is for a dialect to include [common.xml](../messages/common.md) (containing the *MAVLink standard definitions*), extending it with vendor or protocol specific messages.
-While a dialect can include any other message definition, only only a single level of nesting is supported ([at time of writing](https://github.com/ArduPilot/pymavlink/pull/248)).
+While a dialect can include any other message definition, only a single level of nesting is supported ([at time of writing](https://github.com/ArduPilot/pymavlink/pull/248)).
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply doc/mavlink_gitbook.py:47]: Good catch, I will do a upstream PR to fix it.

File: message_definitions/v1.0/all.xml [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+  <!-- matrixpilot.xml: ERROR: Duplicate message id 150 for FLEXIFUNCTION_SET (matrixpilot.xml:50) also used by SENSOR_OFFSETS (ardupilotmega.xml:1101) -->
+  <!-- <include>matrixpilot.xml</include> -->
+  <!-- minimal.xml: duplicates common.xml -->
+  <!-- <include>minimal.xml</include> -->
+  <!-- paparazzi.xml: : ERROR: Duplicate message id 180 for SCRIPT_ITEM (paparazzi.xml:9) also used by CAMERA_FEEDBACK (ardupilotmega.xml:1370) -->
+  <!-- <include>paparazzi.xml</include> -->
+  <!-- python_array_test.xml: ERROR: Duplicate message id 150 for ARRAY_TEST_0 (python_array_test.xml:6) also used by SENSOR_OFFSETS (ardupilotmega.xml:1101) -->
+  <!-- <include>python_array_test.xml</include> -->
+  <!-- slugs.xml: OR: Duplicate message id 170 for CPU_LOAD (slugs.xml:132) also used by DATA32 (ardupilotmega.xml:1276) -->
+  <!-- <include>slugs.xml</include> -->
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply message_definitions/v1.0/all.xml:19]: Again, upstream

File: message_definitions/v1.0/common.xml [hunk 5]
```diff
@@ -2674,7 +2198,8 @@
         <description>Autopilot supports the new param float message type.</description>
       </entry>
       <entry value="4" name="MAV_PROTOCOL_CAPABILITY_MISSION_INT">
-        <description>Autopilot supports MISSION_INT scaled integer message type.</description>
+        <deprecated since="2020-06" replaced_by="">This flag must always be set if missions are supported, because missions must always use MISSION_ITEM_INT (rather than MISSION_ITEM, which is deprecated).</deprecated>
```
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml:2201]: Can you propose a better phrasing upstream?

File: message_definitions/v1.0/uAvionix.xml (hunk 2 above)
Comment (amilcarlucas) [reply message_definitions/v1.0/uAvionix.xml:7]: I will remove this commit, and rebase

**Comment by amilcarlucas** [comment]
@auturgy thanks for the thorough review.

File: message_definitions/v1.0/common.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (amilcarlucas) [reply message_definitions/v1.0/common.xml:2201]: This one has already been merged in by @peterbarker .
Can you propose a better phrasing upstream?

**Comment by amilcarlucas** [comment]
These are the comments from @peterbarker regarding this PR:
```
Need discussion:
  Deprecate_MAV_CMD_REQUEST - moves MAV_CMD_REQUEST_MESSAGE to minimal
  WiFi configuration - dubious changes require in-depth thought (int8_t for an enum?!  Really?!)
  Add-yaw-to-set-home - probably good, but implications we need to discuss
  VFR_HUD comment change - needs discussion
  comment on nav-vtol-takeoff - advice isn't necessarily good; a prearm failure might be much better!
  comment on nav-takeoff - needs discussion
  MAV_CMD metadata updates - need to discuss the gimbal-device-id change
  LOITER_TIME (1383) - requires thought
  distance_sensor (1403) - requires ArduPilot changes
  mav-cmd-units (1105) - removing min/max value from angle for no reason?
  loiter-time (1402) - complicated, needs thought
  DO_PARACHUTE comment changes (1417) - dubious comment changes
  battery status (1423) - need ardupilot changes
  terrain protocol (#1437) - deprecates terrain_check which we use a lot!
  move-mav-cmd-fix-mag-cal-yaw - hasLocation used inappropriately
  deprecate nav-fence-return-point (1480) - we use this for specifying a breach return point for a single fence.  This is still useful.
  format_xml.sh - need to discuss
  spellng-and-small-misc-corrections - we don't support CAMERA_CAPTURE_STATUS
  deprecate some messages - we still use MAV_DATA_STREAM extensively
  improve wording - need to check we handle the pitch/roll/yaw offsets the same.  also need to check the camera-id change
```

**Comment by amilcarlucas** [comment]
CI test in Ardupilot were passing before on ArduPilot/ardupilot#15404 so I expect them to pass again.

**Comment by davidbuzz** [comment]
General concensus on the ardupilot dev call is that this is 'too big' or a PR to easily review/merge and might be better split out into separate ones..?

**Comment by peterbarker** [comment]
I'm going to have another look through this.

**Comment by peterbarker** [comment]
```
commit 9045687be3483c60471e631b485f7dfad41ec93d
Author: olliw42 <6089567+olliw42@users.noreply.github.com>
Date:   Mon Apr 27 03:15:24 2020 +0200

    Deprecate MAV_CMD_REQUEST_* in favor of MAV_CMD_REQUEST_MESSAGE (#1180)
```

I'm not convinced that MAV_CMD_REQUEST_MESSAGE should be in minimal.xml.

Deprecating the messages seems fine.

Split this one into a separate PR we can discuss on a devcall?

**Comment by peterbarker** [comment]
The pymavlink commit in here needs to be removed.

**Comment by peterbarker** [comment]
I've had a look back through these.  My observations already in this PR still apply - the relevant commits are not trivially mergeable as they either require discussion or (IMO) shouldn't be merged.

Please break them out into separate PRs - *some* of them might get in, then, rather than them all hanging around because of some controversial stuff.

I've also split out the 2 new commits which are straight-forward.

**Comment by amilcarlucas** [comment]
Moved parts of this to #156 #157 #158 #159 #160
Once those are discussed and merged I will rebase this one, and solve the conflicts

**Comment by amilcarlucas** [comment]
Now that all non-controversial patches have been moved to to 5 other PRs and/or merged, only 5 controversial commits remain here :)

File: message_definitions/v1.0/common.xml [hunk 6]
```diff
@@ -848,19 +893,19 @@
         <param index="2">Empty</param>
         <param index="3">Empty</param>
         <param index="4">Empty</param>
-        <param index="5">Latitude</param>
-        <param index="6">Longitude</param>
-        <param index="7">Altitude</param>
+        <param index="5" label="Latitude" units="degE7">Latitude of ROI location</param>
```
Comment (tridge) [inline message_definitions/v1.0/common.xml:896]: can't do unit, as varies for COMMAND vs COMMAND_LONG

File: message_definitions/v1.0/common.xml [hunk 7]
```diff
@@ -3418,6 +3464,7 @@
       <field type="uint8_t" name="rssi">Receive signal strength indicator in device-dependent units/scale. Values: [0-254], 255: invalid/unknown.</field>
     </message>
     <message id="66" name="REQUEST_DATA_STREAM">
+      <deprecated since="2015-08" replaced_by="SET_MESSAGE_INTERVAL"/>
```
Comment (tridge) [inline message_definitions/v1.0/common.xml:3467]: remove this deprecation

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by amilcarlucas** [comment]
Requested changes are done

**Comment by peterbarker** [comment]
Merged, thanks @amilcarlucas !

**Pull request merged**